packages/core/saved-objects/core-saved-objects-api-server-internal/src/lib/apis/utils/find_shared_origin_objects.ts

```diff
@@ -80,17 +90,56 @@ export async function findSharedOriginObjects(
   return objectsMap;
 }
 
-function createAliasKueryFilter(objects: Array<{ type: string; origin: string }>) {
+function createOriginKueryFilter(
+  objects: ObjectOrigin[],
+  purpose?: SavedObjectsCollectMultiNamespaceReferencesPurpose
+) {
   const { buildNode } = esKuery.nodeTypes.function;
   // Note: these nodes include '.attributes' for type-level fields because these are eventually passed to `validateConvertFilterToKueryNode`, which requires it
   const kueryNodes = objects
-    .reduce<unknown[]>((acc, { type, origin }) => {
+    .reduce<unknown[]>((acc, { type, id, origin }) => {
       // Escape Kuery values to prevent parsing errors and unintended behavior (object types/IDs can contain KQL special characters/operators)
-      const match1 = buildNode('is', `${type}.id`, esKuery.escapeKuery(`${type}:${origin}`)); // here we are looking for the raw document `_id` field, which has a `type:` prefix
-      const match2 = buildNode('is', `${type}.originId`, esKuery.escapeKuery(origin)); // here we are looking for the saved object's `originId` field, which does not have a `type:` prefix
-      acc.push([match1, match2]);
+
```

Comment:
To reviewers: there are a few ways we can structure this logic. I am not 100% set on the current implementation. Alternatively, we could have an explicit case for `updateObjectsSpaces` that distinctly handles when the shared object has an origin ID, rather than relying on re-using the initial two kuery nodes. This may make it more readable. Let me know what you think.